### ISSUE TITLE
Not working in Dragonflight (Provisional update for 10.0.2 inside)

### ISSUE THREAD

**Issue by tflo** [glassleo/Paragon#4]
In Dragonflight, Paragon does not work and throws this error:

```text
25x Interface/AddOns/Paragon/Core.lua:191: attempt to call global 'GetFriendshipReputation' (a nil value)
[string "@Interface/AddOns/Paragon/Core.lua"]:191: in function <Interface/AddOns/Paragon/Core.lua:180>
[string "@Interface/AddOns/Paragon/Core.lua"]:646: in function <Interface/AddOns/Paragon/Core.lua:634>
```

[According to WoWPedia](https://wowpedia.fandom.com/wiki/API_GetFriendshipReputation), in Dragonflight `GetFriendshipReputation` has been replaced with `C_GossipInfo.GetFriendshipReputation`. 

By changing this (Core.lua, line 191), plus a couple of nil checks for non returned `barMin` values in the same function, I was able to get it to work _provisionally_. Though, I have not investigated further why no values are returned, can be a Blizz bug or another API change.

**Comment by vince2paname** [comment]
Hello tflo, could you please share a link to download the version you have been able to repair ? I really need this addon, thank you !

**Comment by tflo** [comment]
Here the patched Core.lua file: https://d.pr/f/unj6B2 . Just replace the old one inside the Paragon addon folder.

The `C_GossipInfo.GetFriendshipReputation` change is essential and correct due to the API changes, however the 3 added nil checks are provisonal hacks, i.e. they just make the addon working, but they do not eliminate the cause of missing min values.

That being said, _so far_ it is working for me, but I'm using it mainly with Legion rep tokens, maybe it freaks out or gives wrong values with other reputations. Double-check the values the addon returns!

Edit 2022-11-16: **Do not use this patch with 10.0.2. Scroll down a few posts for a 10.0.2-compatible version.**

**Comment by vince2paname** [comment]
Thank you very much, Tflo ! Worked perfectly. I'm also using it mostly for Legion rep.

**Comment by tflo** [comment]
Lol, broken again in 10.0.2. They generously [changed the whole tooltip API](https://wowpedia.fandom.com/wiki/Patch_10.0.2/API_changes#Tooltip_Changes). Looks like this needs some more work than updating a function name…

**Comment by tflo** [comment]
OK, here is a working version for 10.0.2. It's the whole addon this time, as I have updated the toc file too. Use with caution!

[Paragon - Inofficial and hackish patch for 10.0.2 - v2.zip](https://github.com/glassleo/Paragon/files/10024855/Paragon.-.Inofficial.and.hackish.patch.for.10.0.2.-.v2.zip)


Known issues:

~~If you have an item comparison tooltip (eg hovering over a gear item in your bags with the Shift key down, or hovering over a gear item in a merchant frame), then the addon throws a 'GetItem' error. This shouldn't affect the normal functionality though.~~ (If I find the time I will try to fix it, but my plan is not to do a fullscale maintenance of this addon ;)

Fixed (hopefully). Changed download file above to v2.

**Comment by vince2paname** [comment]
Thank you Tflo, really appreciated ! Installed and working.

**Comment by tflo** [comment]
@vince2paname Please try version 2 from above. This should fix the issue with comparison tooltips.

**Comment by vince2paname** [comment]
This version works better. I had some lua errors previously. Thanks again.

**Comment by glassleo** [comment]
Sorry I haven't updated the addon yet. I've been considering a complete rewrite including a better interface.

I'll try to get a working version out soon.

**Comment by glassleo** [comment]
I have now released a working version for 10.0.2: [Chalcedony 4](https://github.com/glassleo/Paragon/releases/)

**Comment by tflo** [comment]
Thank you!
